packages/@ourworldindata/grapher/src/lineCharts/LineChart.tsx

```diff
@@ -481,7 +484,29 @@ export class LineChart
 
         if (!target) return undefined
 
-        const sortedData = sortBy(this.series, (series) => {
+        // Duplicate seriesNames will be present if there is a projected-values line
+        const seriesSegments = mapValues(
+            groupBy(this.series, "seriesName"),
+            (segments) =>
+                segments.find((series) =>
+                    // Ideally pick series with a defined value at the target time
+                    series.points.find((point) => point.x === target.x)
+                ) ??
+                segments.find((series): boolean | void => {
+                    // Otherwise pick the series whose start & end contains the target time
+                    // and display a "No data" notice.
+                    const [startX, endX] = extent(series.points, ({ x }) => x)
+                    return (
+                        isNumber(startX) &&
+                        isNumber(endX) &&
+                        startX < target.x &&
+                        target.x < endX
+                    )
+                }) ??
+                null // If neither series matches, exclude the entity from the tooltip altogether
+        )
+
+        const sortedData = sortBy(compact(values(seriesSegments)), (series) => {
```

Comment:
Ah, I realise now that the values we iterate over here are objects, not raw numbers.
Still, I would change it to `excludeNullish` and prefer that over using `compact` going into the future just because it's so easy to accidentally filter out zero values otherwise.